app/api/domains/osu.py

```diff
@@ -1667,6 +1677,13 @@ async def register_account(
     # are safe for registration.
     errors: Mapping[str, list[str]] = defaultdict(list)
 
+    # Disable in-game registration if enabled
+    if app.settings.DISALLOW_INGAME_REGISTRATION:
+        return ORJSONResponse(
+            content=INGAME_REGISTRATION_DISALLOWED_ERROR,
+            status_code=status.HTTP_400_BAD_REQUEST,
+        )
```

Comment:
might as well move it above `errors` initialization.. the param validation is more debatable ig